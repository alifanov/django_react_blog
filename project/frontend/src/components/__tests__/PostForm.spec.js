import React from "react";
import PostForm from "../forms/Post";

import { shallow, mount } from "../../enzyme";
import { MemoryRouter } from "react-router-dom";

import renderer from "react-test-renderer";

describe(">>> PostForm test", () => {
  let form;
  beforeEach(() => {
    form = shallow(<PostForm />);
  });

  it("+++ render component", () => {
    expect(form.length).toEqual(1);
  });
  it("+++ check elements", () => {
    expect(form.find("Form>FormGroup>Input").length).toEqual(2);
    expect(form.find("Form>div>Button").length).toEqual(1);
  });
  it("+++ check add text", () => {
    const component = form.dive();
    component
      .find("Form>FormGroup>Input")
      .at(0)
      .simulate("change", { target: { value: "New title" } });
    component
      .find("Form>FormGroup>Input")
      .at(1)
      .simulate("change", { target: { value: "New text" } });

    expect(form.state().title).toEqual("New title");
    expect(form.state().text).toEqual("New text");
  });
  it("+++ check show/hide", () => {
    expect(form.state().show).toEqual(false);
    const component = form.dive();
    component
      .find("a")
      .at(0)
      .simulate("click");

    expect(form.state().show).toEqual(true);
  });
  it("+++ check submit", () => {
    const mockSubmitFn = jest.fn();
    form = shallow(<PostForm onSubmit={mockSubmitFn} />);
    const component = form.dive();
    component
      .find("Form>FormGroup>Input")
      .at(0)
      .simulate("change", { target: { value: "New title" } });
    component
      .find("Form>FormGroup>Input")
      .at(1)
      .simulate("change", { target: { value: "New text" } });

    component
      .find("Form>div>Button")
      .at(0)
      .simulate("click");

    expect(form.state().title).toEqual("");
    expect(form.state().text).toEqual("");
    expect(mockSubmitFn.mock.calls.length).toEqual(1);
  });
});

describe(">>>PostForm --- Snapshot", () => {
  it("+++capturing", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <PostForm />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
