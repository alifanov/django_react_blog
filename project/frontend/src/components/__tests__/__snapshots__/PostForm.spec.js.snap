// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>PostForm --- Snapshot +++capturing 1`] = `
<div
  className="transparent p-0 jumbotron"
>
  <a
    href="#"
    onClick={[Function]}
  >
    <h4>
      New post
    </h4>
  </a>
  <form
    className="d-none"
  >
    <div
      className="position-relative form-group"
    >
      <label
        className=""
        htmlFor="exampleTitle"
      >
        Title
      </label>
      <input
        className="form-control"
        id="exampleTitle"
        name="title"
        onChange={[Function]}
        placeholder="Enter post title"
        type="text"
        value=""
      />
    </div>
    <div
      className="position-relative form-group"
    >
      <label
        className=""
        htmlFor="exampleText"
      >
        Text Area
      </label>
      <textarea
        className="form-control"
        id="exampleText"
        name="text"
        onChange={[Function]}
        placeholder="Enter post text"
        value=""
      />
    </div>
    <div
      className="position-relative form-group"
    >
      <div
        className="react-tags"
        onClick={[Function]}
      >
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          className="react-tags__selected"
        />
        <div
          className="react-tags__search"
        >
          <div
            className="react-tags__search-input"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-label="Add new tag"
              aria-owns="ReactTags-listbox"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              placeholder="Add new tag"
              role="combobox"
              style={
                Object {
                  "width": null,
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                  "width": 0,
                }
              }
            >
              Add new tag
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="d-flex justify-content-end"
    >
      <button
        aria-label={null}
        className="btn btn-secondary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Add new post
      </button>
    </div>
  </form>
</div>
`;
